### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by alimJalanTech**
Odin Twitter Clone

## Features

- Sidebar
- Tweetbox
- Tweets Field
- News Feed
- Favorite Tweet

## UI

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,13 +2,14 @@ import React from "react";
 import "./App.css";
 import Sidebar from "./components/Sidebar";
 import Feed from "./components/Feed";
-import Post from "./components/Post";
+import Widget from "./components/Widget";
 
-function App() {
+function App(): JSX.Element {
```
Comment (roopaljasnani) [inline src/App.tsx]: Please use React.FC as return type.

JSX.Element is used when returning snippets under a larger react component

File: src/components/Feed.tsx [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+interface PostData {
+  avatar: string;
+  displayName: string;
+  image: string;
+  text: string;
+  userName: string;
+  verified: boolean;
+}
+
+const Feed: React.FC = (): JSX.Element => {
```
Comment (roopaljasnani) [inline src/components/Feed.tsx]: If you have used React.FC, you don't need to use JSX.Element

File: src/components/Feed.tsx [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
+  const [posts, setPosts] = useState<PostData[]>([]);
+
+  useEffect(() => {
+    const unsubscribe = db.collection("posts").onSnapshot((snapshot) => {
+      const data = snapshot.docs.map((doc) => {
+        return {
```
Comment (roopaljasnani) [inline src/components/Feed.tsx]: use implicit return with arrow functions

```tsx
const data = snapshot.docs.map(
  (doc) =>
    ({
      avatar: doc.data().avatar,
      displayName: doc.data().displayName,
      image: doc.data().image,
      text: doc.data().text,
      userName: doc.data().userName,
      verified: doc.data().verified,
    } as PostData)
);
```

File: src/components/Post.tsx [hunk 4]
```diff
@@ -7,34 +7,50 @@ import ChatBubbleOutlineIcon from "@material-ui/icons/ChatBubbleOutline";
...
         </div>
-        <img
-          src="https://unsplash.com/photos/iEEBWgY_6lA/download?ixid=MnwxMjA3fDB8MXxhbGx8fHx8fHx8fHwxNjgxODQ5ODYy&force=true&w=640"
-          alt="Avatar Icon"
-        />
-        <ChatBubbleOutlineIcon />
-        <RepeatIcon />
-        <FavoriteBorderIcon />
-        <PublishIcon />
+        <img src={image} alt="" />
```
Comment (roopaljasnani) [inline src/components/Post.tsx]: pls always specify alt value on image tags

File: src/components/Sidebar.tsx [hunk 5]
```diff
@@ -12,18 +12,29 @@ import MoreHorizon from "@material-ui/icons/MoreHoriz";
...
+const Sidebar: React.FC = () => {
+  const sidebarOptions: SidebarOption[] = [
+    { Icon: HomeIcon, text: "Home" },
+    { Icon: SearchIcon, text: "Search" },
+    { Icon: NotificationsNoneIcon, text: "Notifications" },
+    { Icon: MailOutlineIcon, text: "Mail" },
+    { Icon: BookmarkBorderIcon, text: "Bookmarks" },
+    { Icon: ListAltIcon, text: "Lists" },
+    { Icon: PermIdentityIcon, text: "Identity" },
+    { Icon: MoreHorizon, text: "More" },
```
Comment (roopaljasnani) [inline src/components/Sidebar.tsx:29]: Did you not need to import these icons?

File: src/components/Tweetbox.tsx [hunk 6]
```diff
@@ -1,21 +1,50 @@
-import React from "react";
+import React, { useState } from "react";
 import { Avatar, Button } from "@material-ui/core";
 import "./Tweetbox.css";
+import db from "../firebase";
 
-const Tweetbox = () => {
+type TweetboxProps = {};
+
+const Tweetbox: React.FC<TweetboxProps> = () => {
```
Comment (roopaljasnani) [inline src/components/Tweetbox.tsx]: If there are no props, use just `React.FC` , TweetboxProps is not needed here

File: src/components/Tweetbox.tsx [hunk 7]
```diff
@@ -1,21 +1,50 @@
...
+  const [tweetMessage, setTweetMessage] = useState<string>("");
+  const [tweetImage, setTweetImage] = useState<string>("");
+
+  const sendTweet = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
```
Comment (roopaljasnani) [inline src/components/Tweetbox.tsx]: return type on function

File: src/components/Widget.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+const Widget: React.FC = () => {
+  return (
+    <div className="widgets">
+      <div className="widgets-input">
+        <Search className="widgets-search-icon" />
+        <input placeholder="Search Twitter" type="text" />
+      </div>
+      <div className="widget-container">
+        <h2>Whats Happening</h2>
+        <TwitterTweetEmbed tweetId={"1550073471283392512"} />
```
Comment (roopaljasnani) [inline src/components/Widget.tsx]: dont use curly braces with static strings

tweetId="1550073471283392512"

File: src/components/Feed.tsx (hunk 2 above)
Comment (alimJalanTech) [reply src/components/Feed.tsx]: Noted, correcting the issue.

File: src/App.tsx (hunk 1 above)
Comment (alimJalanTech) [reply src/App.tsx]: Thanks for pointing out the issue. Corrected

File: src/components/Feed.tsx (hunk 3 above)
Comment (alimJalanTech) [reply src/components/Feed.tsx]: Corrected.

File: src/components/Post.tsx (hunk 4 above)
Comment (alimJalanTech) [reply src/components/Post.tsx]: Resolved

File: src/components/Tweetbox.tsx (hunk 6 above)
Comment (alimJalanTech) [reply src/components/Tweetbox.tsx]: Resolved

File: src/components/Tweetbox.tsx (hunk 7 above)
Comment (alimJalanTech) [reply src/components/Tweetbox.tsx]: Resolved

File: src/components/Sidebar.tsx (hunk 5 above)
Comment (alimJalanTech) [reply src/components/Sidebar.tsx:29]: imported and stored in a list to map over each object. Please let me know how can I improve it @roopaljasnani

Comment (roopaljasnani) [reply src/components/Sidebar.tsx:29]: NVM. I saw the entire file contents just now, saw the imports also. 👍🏻

File: src/components/Widget.tsx (hunk 8 above)
Comment (alimJalanTech) [reply src/components/Widget.tsx]: corrected

**Review by roopaljasnani — APPROVED** [review]
LGTM!

**Pull request merged**